### PR TITLE
[front] feat: add a page listing the Tournesol Talks

### PR DESCRIPTION
**Pull request by bbosson**
Init Talks Page
Recovering talks from a json file

Issue #1493

### CODE REVIEW

**Comment by GresilleSiffle** [comment]
Hello @bbosson 

I'll take a look at the PR either this afternoon or tomorrow, sorry for the delay.

File: frontend/src/pages/talks/TalkEntryList.tsx
Comment (GresilleSiffle) [inline frontend/src/pages/talks/TalkEntryList.tsx:11]: The horizontal margin is not required here, as the parent component already has an horizontal padding to display the content nicely on each screen size.

```suggestion
        <Box key={talk.uid} my={2}>
```

File: frontend/src/services/mocks/talks-en.json
Comment (GresilleSiffle) [inline frontend/src/services/mocks/talks-en.json:4]: To ease the navigation, let's also add the key `name`.

```suggestion
     "name": "introduction_to_tournesol",
     "title": "Introduction to Tournesol",
```

This key doesn't contain space and will allow us to create HTML anchors. For instance:

 `https://tournesol.app/talk#introduction_to_tournesol` will refer to the Introduction to Tournesol talk.

File: frontend/src/pages/talks/TalkEntryList.tsx
Comment (GresilleSiffle) [inline frontend/src/pages/talks/TalkEntryList.tsx:12]: Let's use the `key={talk.name}` here (see my comment in the JSON file.

File: frontend/src/pages/talks/TalkSingleEntry.tsx
Comment (GresilleSiffle) [inline frontend/src/pages/talks/TalkSingleEntry.tsx:8]: The `compact` prop doesn't seem to be used here.

Should we remove it?

File: frontend/src/pages/talks/TalkSingleEntry.tsx
Comment (GresilleSiffle) [inline frontend/src/pages/talks/TalkSingleEntry.tsx:44]: I see a possible error here. It's possible that the talk is past without a YouTube link if no one took the time to upload the video.

A more robust way could be:
- if the talk is in the future AND there is an invitation link, we display the invitation link
- if the talk is in the past AND there is a YouTube link, we display the YouTube link
- else, no link

What do you think @bbosson ?

File: frontend/src/pages/talks/TalkSingleEntry.tsx
Comment (GresilleSiffle) [inline frontend/src/pages/talks/TalkSingleEntry.tsx:42]: We talked about it during the Thursday meeting but after having used the interface, I realized that wrapping the whole content of the Tournesol Talk entry make it very hard to copy the abstract.

I think copying texts on the internet is a very common operation, and I'm not sure if it's a good idea to also make the abstract clickable.

For instance, I may want to copy the abstract and paste it on Discord.

What do you think @tournesol-app/maintainers  @lenhoanglnh ?

File: frontend/src/pages/talks/TalkSingleEntry.tsx
Comment (GresilleSiffle) [inline frontend/src/pages/talks/TalkSingleEntry.tsx:21]: It looks like we are display this title inside another Typography component.

I suggest to delete this variable and to simply display the `talk.title` inside a single Typography.

File: frontend/src/pages/talks/TalkSingleEntry.tsx
Comment (GresilleSiffle) [inline frontend/src/pages/talks/TalkSingleEntry.tsx:46]: There is already a margin bottom defined in the `TalkEntryList` for each element, I think we can freely remove this `sx`.

File: frontend/src/pages/talks/TalkSingleEntry.tsx
Comment (GresilleSiffle) [inline frontend/src/pages/talks/TalkSingleEntry.tsx:106]: There is a missing `$` here.

 ```suggestion
                 key={`${talk.title}_p${index}`}
```

I propose also to replace `talk.title` by `talk.name`.

File: frontend/src/pages/talks/TalkSingleEntry.tsx
Comment (GresilleSiffle) [inline frontend/src/pages/talks/TalkSingleEntry.tsx:110]: If we use the prop `paragraph` we don't need to specify a margin bottom anymore. It also creates a consistent gap across all paragraphs of the app.

```jsx
{/* use this */}
<Typography paragraph></Typography>

{/* instead of  */}
<Typography sx={{marginBottom: '10px'}}></Typography>
```

File: frontend/src/pages/talks/Talks.tsx
Comment (GresilleSiffle) [inline frontend/src/pages/talks/Talks.tsx:54]: Here we don't need to specify the setter `setUpcomingTalks` and `setPastTalks`, we can safely remove them.

File: frontend/src/services/mocks/TalkEntry.ts
Comment (GresilleSiffle) [inline frontend/src/services/mocks/TalkEntry.ts:7]: Do we want those two attribute to be possibly `undefined`?

I think `youtube_link` should  be `youtube_link?` so we are not forced to define a YouTube link before the upload of a video.

We may also generate the invitation link later, so I think the rest of the code should be a bit refactored to work even with `invitation_link?`.

File: frontend/src/utils/types.ts
Comment (GresilleSiffle) [inline frontend/src/utils/types.ts:48]: Little typo here `TASKS` vs `TALKS`.

**Review by GresilleSiffle — COMMENTED** [review]
Thanks you for your work @bbosson 

The page looks really nice :+1: 

I made few suggestions to harmonize the code with the rest of the app. Do not hesitate if you have any question. Thank you very much.

Before merging I'll have a second look at the CSS to see what can be simplified (like overlapping margin and so on).

File: frontend/src/utils/types.ts
Comment (bbosson) [reply frontend/src/utils/types.ts]: it's fixed

File: frontend/src/services/mocks/TalkEntry.ts
Comment (bbosson) [reply frontend/src/services/mocks/TalkEntry.ts]: it's specified and the impacts achieved

File: frontend/src/pages/talks/Talks.tsx
Comment (bbosson) [reply frontend/src/pages/talks/Talks.tsx]: remove

File: frontend/src/pages/talks/TalkSingleEntry.tsx
Comment (bbosson) [reply frontend/src/pages/talks/TalkSingleEntry.tsx]: it's changed, thanks

File: frontend/src/pages/talks/TalkSingleEntry.tsx
Comment (bbosson) [reply frontend/src/pages/talks/TalkSingleEntry.tsx]: it's changed, thanks

File: frontend/src/services/mocks/talks-en.json
Comment (bbosson) [reply frontend/src/services/mocks/talks-en.json:4]: good idea, it's done

File: frontend/src/pages/talks/TalkEntryList.tsx
Comment (bbosson) [reply frontend/src/pages/talks/TalkEntryList.tsx]: it's remove, thanks

File: frontend/public/images/coming-soon.jpg
Comment (bbosson) [reply frontend/public/images/coming-soon.jpg:1]: I got this image here:

https://www.freepik.com/free-vector/coming-soon-background-with-megaphone-design_10016638.htm

The license says :

Our license allows you to use the content
-     For commercial and personal projects
-     On digital or printed media
-     For an unlimited number of times and without any time limits
-     From anywhere in the world
-     To make modifications and create derivative works

How to attribute?
For example: websites, social media, blogs, ebooks, newsletters, etc.

Copy this link and paste it in a visible place, close to where you’re using the image. If that’s not possible, place it at the footer of your website, blog or newsletter, or in the credits section. 

Where to put this link on the website or do we leave on a Tournesol image?

File: frontend/src/pages/talks/TalkSingleEntry.tsx
Comment (bbosson) [reply frontend/src/pages/talks/TalkSingleEntry.tsx:8]: Yes, It's removed.

File: frontend/src/pages/talks/TalkEntryList.tsx
Comment (bbosson) [reply frontend/src/pages/talks/TalkEntryList.tsx:12]: Good idea now that we have this attribute, it's changed

File: frontend/src/pages/talks/TalkSingleEntry.tsx
Comment (bbosson) [reply frontend/src/pages/talks/TalkSingleEntry.tsx]: Yes, my bad

File: frontend/public/images/coming-soon.jpg
Comment (bbosson) [reply frontend/public/images/coming-soon.jpg:1]: So, following your advice, I removed the image to put the one on the home page.

**Comment by bbosson** [comment]
@lfaucon @GresilleSiffle One question I'm still wondering about is: Should I display an empty section and therefore the title "Upcoming" or "Past" if there are no scheduled or recorded talks? I haven't set any conditions on this.

**Comment by GresilleSiffle** [comment]
I'd say we can display an empty upcoming section to inform the users that we plan events and they can see the upcoming events on this page.

The upcoming title following by a simple paragraph should do the trick 

```
title: Upcoming

paragraph: No Talks are planned in the near future.
```

(In the future we may want to also add "Follow us on Twitter and on Discord do not miss anything".)

For the past section I think it's not required to display it when there is no past talks.

**Comment by bbosson** [comment]
> I'd say we can display an empty upcoming section to inform the users that we plan events and they can see the upcoming events on this page.
> 
> The upcoming title following by a simple paragraph should do the trick
> 
> ```
> title: Upcoming
> 
> paragraph: No Talks are planned in the near future.
> ```
> 
> (In the future we may want to also add "Follow us on Twitter and on Discord do not miss anything".)
> 
> For the past section I think it's not required to display it when there is no past talks.

I made this change

**Comment by GresilleSiffle** [comment]
**review**
- [ ] use the word Events / Évènement for a slightly better SEO
  - [ ] make them plural only when required 
- [ ] the sections' titles should not be h6 if they contain h4 headings

**Comment by GresilleSiffle** [comment]
I'm near the end of the review : ) I just need to do more tests on mobile and to review a bit the code of the `<TalkImagery>` and we are good. I mean, we also need to update the hard coded data with the correct talks (at least few of them).

**Comment by GresilleSiffle** [comment]
I think we want also a short paragraphs that:
- introduces the Tournesol Talks before listing them ("we regularly invite experts to talk about this and that")
- invites the user to subscribe to the mailing list with a nice MUI button

**Comment by GresilleSiffle** [comment]
Here is a preview of the page.

In its current state, I think it messes some information:
- [ ] name and title of the speakers
- [ ] a button "replay" in the bottom right corner of each box
- [ ] the abstract of "Platform Design and Governance for Social Justice"

In the future we could:
- at the bottom of each Talk, add a link to a research paper if any
- make the YouTube link, a Tournesol link instead

screencapture-localhost-3000-talks-2023-06-12-16_00_05

**Comment by GresilleSiffle** [comment]
I added the name of the speaker before each abstract.

What do you think @lenhoanglnh @aidanjungo @lfaucon @amatissart @bbosson ?

screencapture-localhost-3000-talks-2023-06-13-14_58_46

**Comment by lfaucon** [comment]
> I added the name of the speaker before each abstract.
> What do you think @lenhoanglnh @aidanjungo @lfaucon @amatissart @bbosson ?

Looks very good to me. We definitely must put forward the speacked name. Another option may be in the title bar with the format "<speacker> - <title>"

**Comment by GresilleSiffle** [comment]
I'm currently collecting the title of each speaker to finish the page.

**Comment by aidanjungo** [comment]
I haven't read the code but I tested the PR, it looks nice, my only small comment is that we could remove "Abstract: " it is not very useful as we just have the abstract after the title and the name.

**Review by GresilleSiffle — APPROVED** [review]

**Pull request merged**
